tests/basic/volume_ops_test.go

```diff
@@ -3240,3 +3240,195 @@ func writeFioDataToVolume(volName string, n node.Node, size int64) error {
+						    3. create a volume with max of target size of the pool
+						    4. Now update the volume size more than size of the storage pool
+						    5. Check vol size is successful or not
+						    6. Also validate the creation of volume with size more than available capacity on the node
+
+
+	*/
+	JustBeforeEach(func() {
+		StartTorpedoTest("OverCommitVolumeTest", "Validate Overcommit volume size", nil, 0)
+	})
+	// check the size left in the node
+	itLog := "honor OverCommitPercent when resizing the volume"
+	It(itLog, func() {
+		getRandomPoolandCalculateSize := func(snapshotPercent uint64) (selectedNode *node.Node, targetSizeGiB uint64) {
+			stNodes := node.GetStorageDriverNodes()
+			index := rand.Intn(len(stNodes))
+			selectedNode = &stNodes[index]
+			pools := selectedNode.Pools
+			poolToResize := pools[rand.Intn(len(pools))]
+			poolIDToResize := poolToResize.Uuid
+			originalSizeInBytes := poolToResize.TotalSize
+			log.InfoD("Original size of the pool %s is %d", poolIDToResize, originalSizeInBytes)
+			SnapshotPercent := snapshotPercent
+			SubtractSize := (SnapshotPercent * originalSizeInBytes) / 100
+			targetSizeInBytes := originalSizeInBytes - SubtractSize
+			targetSizeGiB = targetSizeInBytes / units.GiB
+			log.InfoD("Target size of the pool %s is %d", poolIDToResize, targetSizeGiB)
+			return selectedNode, targetSizeGiB
+		}
+		CreateVolumeandValidate := func(selectedNode *node.Node, multiple uint64, targetSizeGiB uint64) {
+			id := uuid.New()
+			VolName := fmt.Sprintf("volume_%s", id.String()[:8])
+			log.InfoD("Create a volume with a min size on node [%s]", selectedNode.Name)
+			basicVolumeCreate := fmt.Sprintf("volume create --nodes %s %s", selectedNode.Id, VolName)
+			_, err := runPxctlCommand(basicVolumeCreate, *selectedNode, nil)
+			log.FailOnError(err, "volume creation failed on the cluster with volume name [%s]", VolName)
+			log.InfoD("Base Volume creation with volume name %s successful", VolName)
+			log.InfoD("Now Resize volume with a size of %d %d time of targetsize of pool on node [%s] as %d overcommit percent imposed", multiple*targetSizeGiB, multiple, selectedNode.Name, multiple*100)
+			resizeVolumeCmd := fmt.Sprintf("volume update --size %d %s", multiple*targetSizeGiB, VolName)
+			_, volCreateErr := runPxctlCommand(resizeVolumeCmd, *selectedNode, nil)
+			log.FailOnError(volCreateErr, "volume resize failed  on the cluster with volume name [%s]", VolName)
+			log.InfoD("Volume [%s] resized to %d GiB", VolName, multiple*targetSizeGiB)
+			log.InfoD("Resize the volume more than %d times available capacity on the node [%s]", multiple, selectedNode.Name)
+			resizeVolumeGreaterThanPoolSizeCmd := fmt.Sprintf("volume update --size %d %s", (multiple+1)*targetSizeGiB, VolName)
+			_, err = runPxctlCommand(resizeVolumeGreaterThanPoolSizeCmd, *selectedNode, nil)
+			if err != nil {
+				if strings.Contains(err.Error(), "Failed to resize volume") {
```

Comment:
if u have different error fail it instead of log.InfoD make dash verification so it is not expected, it will fail